client/grpc/tx.go

```diff
@@ -0,0 +1,126 @@
+package client
+
+import (
+	"context"
+	"github.com/bnb-chain/gnfd-go-sdk/types"
+	clitx "github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	xauthsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	"google.golang.org/grpc"
+)
+
+type TransactionClient interface {
+	BroadcastTx(msgs []sdk.Msg, txOpt *types.TxOption, opts ...grpc.CallOption) (*types.TxBroadcastResponse, error)
+	SendToken(req types.SendTokenRequest, txOpt *types.TxOption, opts ...grpc.CallOption) (*types.TxBroadcastResponse, error)
+}
+
+func (c *GreenfieldClient) BroadcastTx(msgs []sdk.Msg, txOpt *types.TxOption, opts ...grpc.CallOption) (*types.TxBroadcastResponse, error) {
+
+	txConfig := authtx.NewTxConfig(types.Cdc(), authtx.DefaultSignModes)
+	txBuilder := txConfig.NewTxBuilder()
+
+	err := txBuilder.SetMsgs(msgs...)
+	if err != nil {
+		return nil, err
+	}
+
+	txBuilder.SetGasLimit(types.DefaultGasLimit)
```

Comment:
I don't think `DefaultGasLimit` is good idea, we should simulate the tx and get suggested gas limit and then broadcast.

Reply:
fixed by using simulated gas